bin/p2-dsctl/main.go

```diff
@@ -129,7 +131,14 @@ func main() {
 			log.Fatalf("Error occurred: %v", err)
 		}
 
-		ds, err := dsstore.Create(manifest, minHealth, name, selector, podID, *createTimeout)
+		ctx, cancelFunc := transaction.New(context.Background())
+		defer cancelFunc()
+		ds, err := dsstore.Create(ctx, manifest, minHealth, name, selector, podID, *createTimeout)
```

Comment:
right now the advantage is the deferred creation of the daemon set. `Create()` creates a `DeamonSet` which is required to pass to `DSContends`, but the problem is i don't want the creation to succeed until i know there isn't contention.

I agree that there are ways to accomplish this without using a transaction but i figured we probably want this to be transactional soon anyway, i think audit log records for daemon set creations will be nice